src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts

```diff
@@ -122,59 +125,51 @@ function useSidebarMetadataFetcher(
         [onApiError],
     );
 
-    const deleteMetadataInstanceSuccessCallback = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
-            const updatedInstances = templateInstances.filter(
-                templateInstance =>
-                    templateInstance.scope !== metadataInstance.scope &&
-                    templateInstance.templateKey !== metadataInstance.templateKey,
-            );
-            setTemplateInstances(updatedInstances);
-        },
-        [templateInstances],
-    );
-
     const handleDeleteMetadataInstance = React.useCallback(
-        (metadataInstance: MetadataTemplateInstance) => {
+        async (metadataInstance: MetadataTemplateInstance): Promise<void> => {
             if (!file || !metadataInstance) {
                 return;
             }
-
-            api.getMetadataAPI(false).deleteMetadata(
-                file,
-                metadataInstance,
-                deleteMetadataInstanceSuccessCallback,
-                onApiError,
-                true,
-            );
+            setStatus(STATUS.LOADING);
+            await api
+                .getMetadataAPI(false)
+                .deleteMetadata(file, metadataInstance, () => setStatus(STATUS.SUCCESS), onApiError, true);
```

Comment:
❓ seems like this is now just setting that status to success instead of updating the template instances. We no longer need to update template instances?

Reply:
`templateInstances` are linked to an array in api cache and are updated in the Metadata.js